Woche2/substitution/substitution.c

```diff
@@ -0,0 +1,65 @@
+#include <cs50.h>
+#include <stdio.h>
+#include <string.h>
+#include <ctype.h>
+
+int main(int argc, string argv[]) {
+
+    int count_key_characters = 0;
+    int count_repeated_characters = 0;
+
+    if (argc != 2) {
+        printf("Usage: ./substitution key\n");
+        return 1;
+    }
+
+    if(strlen(argv[1]) != 26) {
+        printf("Key must contain 26 characters.\n");
+        return 1;
+    }
+
+    for (int i = 0, n = strlen(argv[1]); i < n; i++) {
+        bool CurrentLetterIsAlphabetic = isalpha(argv[1][i]);
+        if(CurrentLetterIsAlphabetic) {
+            count_key_characters++;
+        } else if(count_key_characters != strlen(argv[1])) {
+            printf("Key must only contain alphabetic characters.\n");
+            return 1;
+        }
+    }
+
+    for(int j = 0, n = strlen(argv[1]); j < n; j++) {
+        for(int k = j + 1; k < n; k++) {
+            if(tolower(argv[1][j]) == tolower(argv[1][k])) {
+                count_repeated_characters++;
+            }
+            
+        }
+        
+        if(count_repeated_characters != 0) {
+                printf("Key must not contain repeated characters.\n");
+                return 1;
+        }
+    }
+
+    string plain = get_string("Plainetext: ");
+ 
+    printf("ciphertext: ");
+    for(int w = 0, x = strlen(plain); w < x; w++) {
+        bool CharacterIsUppercase = isalpha(plain[w]) && isupper(plain[w]);
```

Comment:
```suggestion
        bool CharacterIsUppercase =  isupper(plain[w]);
```
You already checked before if you have only alphabetic characters, making this check redundant.